pkg/util/errors.go

```diff
@@ -186,3 +192,145 @@ func HandleStorageError(err error, entityName string) error {
+		return err
+	}
+
+	switch err {
+	case ErrCatalogUsesReservedPlanName:
+		return &HTTPError{
+			ErrorType:   "BadRequest",
+			Description: fmt.Sprintf("The plan name you used for registration \"%s\" is reserved for the Service Manager; you must choose a different name.", entityName),
+			StatusCode:  http.StatusBadRequest,
+		}
+	case ErrPlanMustBeBindable:
+		return &HTTPError{
+			ErrorType:   "BadRequest",
+			Description: fmt.Sprintf("The plan %s must be defined as 'bindable' so that it can support instance sharing.", entityName),
+			StatusCode:  http.StatusBadRequest,
+		}
+	case ErrReferencedInstanceNotShared:
+		return &HTTPError{
+			ErrorType:   "BadRequest",
+			Description: fmt.Sprintf("Failed to create the reference. The instance %s, for which you want to create the reference, must be shared first.", entityName),
+			StatusCode:  http.StatusBadRequest,
+		}
+	case ErrChangingPlanOfReferenceInstance:
+		return &HTTPError{
+			ErrorType:   "BadRequest",
+			Description: fmt.Sprintf("Failed to update the instance %s. This is a reference instance, therefore its plan can't be changed.", entityName),
+			StatusCode:  http.StatusBadRequest,
+		}
+	case ErrChangingPlanOfSharedInstance:
+		return &HTTPError{
+			ErrorType:   "BadRequest",
+			Description: fmt.Sprintf("Couldn't update the instance's plan. The instance %s is shared, therefore you must unshare it first.", entityName),
+			StatusCode:  http.StatusBadRequest,
+		}
+	case ErrChangingParametersOfReferenceInstance:
+		return &HTTPError{
+			ErrorType:   "BadRequest",
+			Description: fmt.Sprintf("Failed to update the instance %s. This is a reference instance, therefore its parameters can't be changed.", entityName),
+			StatusCode:  http.StatusBadRequest,
+		}
+	case ErrMissingOrInvalidReferenceParameter:
+		return &HTTPError{
+			ErrorType:   "InvalidRequest",
+			Description: fmt.Sprintf("Failed to create the instance. Missing or invalid parameter \"%s\".", entityName),
+			StatusCode:  http.StatusBadRequest,
+		}
+	case ErrParsingNewCatalogWithReference:
+		return &HTTPError{
+			ErrorType:   "InvalidRequest",
+			Description: fmt.Sprintf("Parsing failed. Couldn't generate the new catalog with the plan \"%s\".", entityName),
+			StatusCode:  http.StatusBadRequest,
+		}
+	case ErrUnknownOSBMethod:
+		return &HTTPError{
+			ErrorType:   "BadRequest",
+			Description: fmt.Sprintf("Unknown OSB method: \"%s\".", entityName),
+			StatusCode:  http.StatusBadRequest,
+		}
+	case ErrSharedPlanHasReferences:
+		errorMessage := fmt.Sprintf("Couldn't update the service plan. Before you can set it as supportInstanceSharing=false, you first need to unshare the instances of the plan: %s.", entityName)
+		return &HTTPError{
+			ErrorType:   "BadRequest",
+			Description: errorMessage,
+			StatusCode:  http.StatusBadRequest,
+		}
+	case ErrInvalidShareRequest:
+		errorMessage := fmt.Sprintf("Couldn't set the shared property of the instance %s with other changes at the same time.", entityName)
```

Comment:
Couldn't set the property of the instance to "shared". When modifying the "shared" property, you can't add other properties to modify in that same request. Split your request to one that contains only the "shared" property and the other one that contains all the rest.

Reply:
i think it's implicit that he should "Split your request into one that contains only the "shared" property and the other one that contains all the rest.", and therefor we can remove that part.
also, i think it much preferred to return short messages and this one is really long. 
wdyt?

Reply:
@nirbenrey  that sentence is basically "dummy-proof", I am not sure who will read the error message, are you sure it's implicit for everyone who gets them?
I wanted to ask you that myself but decided to just leave it.

Reply:
Couldn't update the instance "%s". Modifying the "shared" property should be in a dedicated request that doesn't contain any other properties.

Reply:
@nirbenrey  pls use this last version of the error message. 
That one is final.

Reply:
Currently, we have:

`"Couldn't set the property of the instance to \"shared\". When modifying the \"shared\" property, you can't add other properties to modify in that same request. Split your request into one that contains only the \"shared\" property and the other one that contains all the rest."`

is it the final string or we change it?

Reply:
Couldn't update the instance "%s". Modifying the "shared" property should be in a dedicated request that doesn't contain any other properties.

@EpsilonTal  the one I wrote in this comment is the final version.

Reply:
changed to 
`Couldn't update the instance %s. Modifying the "shared" property should be in a dedicated request that doesn't contain any other properties.`

Reply:
@EpsilonTal  yes, that's it.